tests/identify_test.py

```diff
@@ -217,6 +217,13 @@ def test_file_is_text_does_not_exist(tmpdir):
         (b"#!/path'with/quotes    y", ("/path'with/quotes", 'y')),
         # Don't regress on leading/trailing ws
         (b"#! /path'with/quotes y ", ("/path'with/quotes", 'y')),
+        # Test nix-shell specialites with shebang on second line
+        (b'#! /usr/bin/env nix-shell\n#! nix-shell -i bash -p python', ('bash',)),
+        (b'#! /usr/bin/env nix-shell\n#! nix-shell -i python -p coreutils', ('python',)),
+        (b'#! /usr/bin/env nix-shell\n#! nix-shell -p coreutils -i python', ('python',)),
+        (b'#! /usr/bin/env nix-shell', ('nix-shell',)),
+        # edge case - normally nix-shell is on an opaque nix path and accessed through env
+        (b'#! /usr/bin/nix-shell\n#! nix-shell -p coreutils -i python', ('python',)),
```

Comment:
Some more test cases for multi-line, no space after `#!` and quotes.

```suggestion
        (b'#! /usr/bin/env nix-shell\n#! nix-shell -p coreutils -i python', ('python',)),
        # multi-line and no whitespace variation
        (b'#! /usr/bin/env nix-shell\n#! nix-shell -p coreutils\n#! nix-shell -i python', ('python',)),
        (b'#! /usr/bin/env nix-shell\n#!nix-shell -p coreutils\n#!nix-shell -i python', ('python',)),
        # interpret quotes correctly
        (b'#!/usr/bin/env nix-shell\n#!nix-shell --argstr x "a -i python3 p"\n#!nix-shell -p hello\n#!nix-shell -i bash\n#!nix-shell --argstr y "b -i runhaskell q"', ('bash',)),
```

nix-shell supports quotes. Here's [Nix's implementation](https://github.com/NixOS/nix/blob/8a945d6ddb0676b454458e6fe0e9ea6f8b4b5659/src/nix-build/nix-build.cc#L26) which claims to be inspired by Perl's `shellwords`.
Although python's [`shellwords`](https://github.com/mozillazg/python-shellwords/blob/master/shellwords/__init__.py#L43) package seems like a match, with `parse_env` on and `parse_backtick` off. Nix seems to allow env interpolation somewhere after `shellwords`.

Reply:
Ok, I forgot about the multi-line case, will implement it.